### ISSUE TITLE
"HKDF-Extract(0, Z)"

### ISSUE THREAD

**Issue by sayrer** [tlswg/draft-ietf-tls-esni#188]
A second point in issue #155 wasn't addressed. 

"...HKDF-Extract(0, Z) should probably become HKDF-Extract("", Z) or be explicit about the salt as is done in https://tools.ietf.org/html/rfc8446#section-7.1 (or https://tools.ietf.org/html/rfc5869#section-2.2)."

The current draft does clarify this in the text below, but I initially missed that. The reader has to know that the first argument to HKDF-Extract is named "salt" to put it together.

I did end up getting it right, figuring this was intended to mean a buffer of zeros. I think it would be clearer to say:

```
zeros = uint8[Hash.length]
Zx = HKDF-Extract(zeros, Z)
```

and change the paragraph below a bit.

**Comment by chris-wood** [comment]
@sayrer a PR to clarify this point is most welcome!

**Comment by sayrer** [comment]
It would be ok to take this suggestion into consideration under the terms of the IETF's "Note Well" blue sheet terms, and incorporate it into a PR by the editors.

**Comment by chris-wood** [comment]
@sayrer I agree! Could you please provide that PR?

**Comment by sayrer** [comment]
Oh, I was asking you folks to do it. If there's a reason I need to do it, I'd be happy to.

**Comment by ekr** [comment]
I don't think this is necessary, so while I'm willing to look at a PR, I'm not going to make one at this time.
